File: src/main/kotlin/com/stannis/dataModel/statementTypes/QualifiedName.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.NameInterface
import com.stannis.dataModel.Statement

data class QualifiedName(
    override val `$type`: String? = "QualifiedName",
    var qualifier: ArrayList<NameInterface>?,
    var lastName: NameInterface?
) : Statement, NameInterface {
    fun addQualifier(statement: Statement) {
        if (qualifier == null) {
            qualifier = ArrayList()
        }
        qualifier!!.add(statement as NameInterface)
    }

    override fun getWrittenName(): String {
        return lastName!!.getWrittenName()
    }
}

File: Interpreter/Interpreter/Models/serialize/statementTypes/QualifiedName.cs
using System.Collections.Generic;

namespace Interpreter.Models.serialize.statementTypes{
 
    public class QualifiedName : IStatement, INameInterface
    {
        public IList<IStatement> qualifier = new List<IStatement>();
        
        public IStatement lastName { set; get; }


        public string GetWrittenName()
        {
            if (lastName is INameInterface)
            {
                return (lastName as INameInterface).GetWrittenName();
            }
            return "";
        }
    }
}